evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -0,0 +1,141 @@
+import com.here.ort.model.LicenseFinding
+import com.here.ort.model.Package
+import com.here.ort.model.config.Excludes
+import com.here.ort.model.config.PathExclude
+
+/**
+ * A [Rule] to check a single [Package].
+ */
+open class PackageRule(
+    ruleSet: RuleSet,
+    name: String,
+
+    /**
+     * The [Package] to check.
+     */
+    val pkg: Package,
+
+    /**
+     * The detected licenses for the [Package].
+     */
+    val detectedLicenses: List<LicenseFinding>
+) : Rule(ruleSet, name) {
+
+    private val licenseRules = mutableListOf<LicenseRule>()
+
+    override val description = "Evaluating rule '$name' for package '${pkg.id.toCoordinates()}'."
+
+    override fun runInternal() {
+        licenseRules.forEach { it.run() }
+    }
+
+    /**
+     * A [RuleMatcher] that checks if the [package][pkg] has any concluded, declared, or detected license.
+     */
+    fun hasLicense() =
+        object : RuleMatcher {
+            override val description = "hasLicense()"
+
+            override fun matches() =
+                pkg.concludedLicense?.licenses()?.isNotEmpty() == true
+                        || pkg.declaredLicenses.isNotEmpty()
+                        || detectedLicenses.isNotEmpty()
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [package][pkg] is [excluded][Excludes].
+     */
+    fun isExcluded() =
+        object : RuleMatcher {
+            override val description = "isExcluded()"
+
+            override fun matches() = ruleSet.ortResult.isExcluded(pkg.id)
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [identifier][Package.id] of the [package][pkg] belongs to one of the provided
+     * [orgs][Identifier.isFromOrg].
+     */
+    fun isFromOrg(vararg names: String) =
+        object : RuleMatcher {
+            override val description = "isFromOrg(${names.joinToString()})"
+
+            override fun matches() = pkg.id.isFromOrg(*names)
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [identifier type][Identifier.type] of the [package][pkg] equals [type].
+     */
+    fun isType(type: String) =
+        object : RuleMatcher {
+            override val description = "isType($type)"
+
+            override fun matches() = pkg.id.type == type
+        }
+
+    /**
+     * DSL function to configure a [LicenseRule] and add it to this rule.
+     */
+    fun licenseRule(name: String, licenseView: LicenseView, block: LicenseRule.() -> Unit) {
+        val licenses = licenseView.licenses(pkg, detectedLicenses.map { it.license })
+
+        licenses.forEach { (license, licenseSource) ->
+            val findings = if (licenseSource == LicenseSource.DETECTED) {
+                ruleSet.ortResult.collectLicenseFindings()[pkg.id].orEmpty()
+            } else {
+                emptyMap()
+            }
+
+            val licenseRule = LicenseRule(name, license, licenseSource, findings)
+            licenseRule.block()
+
+            licenseRules += licenseRule
```

Comment:
You probably don't like it, but I'm going to say it anyway 😉 You could write this as

    licenseRules += LicenseRule(name, license, licenseSource, findings).apply { block() }

Reply:
Done.